private/highscoreList.js

```diff
@@ -0,0 +1,95 @@
+const router = require('express').Router();
+const db = require('../database/connection.js').connection;
+
+router.get('/api/highscores/user/:userHighestScore', (req, res) => {
+  db.query(`(SELECT high_score.high_score_id, player.username, high_score.score, high_score.date_time 
+        FROM player JOIN high_score 
+        ON player.player_id = high_score.player_id 
+        WHERE score >= ?
+        ORDER BY score ASC
+        LIMIT 5)
+    
+    UNION ALL
+    
+    (SELECT high_score.high_score_id, player.username, high_score.score, high_score.date_time 
+        FROM player JOIN high_score 
+        ON player.player_id = high_score.player_id
+        WHERE score < ?
+        ORDER BY score DESC
+        LIMIT 5);`, [req.params.userHighestScore, req.params.userHighestScore], (error, result, fields) => {
+    if (result && result.length) {
+      const highscores = [];
+      for (const highscore of result) {
+        highscores.push({
+          place: highscore.high_score_id, username: highscore.username, score: highscore.score, dateTime: highscore.date_time,
+        });
+      }
+      res.send({
+        highscores,
+      });
+    } else {
+      res.send({
+        message: 'No scores found',
+      });
+    }
+  });
+});
+
+router.get('/api/highestscore/user/:userId', (req, res) => {
+  db.query('SELECT score FROM high_score WHERE player_id = ? ORDER BY score DESC LIMIT 1;', [req.params.userId], (error, result, fields) => {
+    if (result && result.length) {
+      res.send({
+        highestscore: result[0].score,
+      });
+    } else {
+      res.send({
+        message: 'No scores found.',
+      });
+    }
+  });
+});
+
+router.get('/api/highestscores/:currentPage', (req, res) => {
+  db.query('SELECT * FROM high_score;',(error, result, fields) => {
```

Comment:
I'm trying to think whether this can be done in more efficient way. I get what you're doing but querying for all the highscores and then just not using them seems a bit wasteful. 
One thing is that it should just show top 100 scores and we don't need to store the rest so if we got like idk 10 for page then there would be 10 pages but there would still be issue at the beginning of not all 100 scores are there. This is probably something we will need to look into more.

Reply:
well I implemented it bc we decided to show all the results and I need to know how many records we have in db in order to display right pages

Reply:
count instead of select all